services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/BandwidthSpec.groovy

```diff
@@ -162,4 +162,53 @@ class BandwidthSpec extends BaseSpecification {
         cleanup: "Delete created flow"
         flow?.id && northboundService.deleteFlow(flow.id)
     }
+
+    def "Able to update bandwidth to maximum link speed without using alternate links"() {
+        given: "Two active neighboring switches"
+        def switches = topology.getActiveSwitches()
+        def links = northboundService.getAllLinks()
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.find { Switch src, Switch dst ->
+            links.any { link ->
+                def switchIds = link.path*.switchId
+                switchIds.contains(src.dpId) && switchIds.contains(dst.dpId)
+            }
+        }
+        assert srcSwitch && dstSwitch
+
+        when: "Create a flow with a valid small bandwidth"
+        def maximumBandwidth = 1000
+
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.maximumBandwidth = maximumBandwidth
+        northboundService.addFlow(flow)
+
+        then: "Flow is really created and has 'Up' status"
+        Wrappers.wait(5) {
+            northboundService.getFlowStatus(flow.id).status == FlowState.UP
+        }
+
+        and: "Only one link is involved in flow path"
+        def flowPath = PathHelper.convert(northboundService.getFlowPath(flow.id))
+        def involvedLinks = pathHelper.getInvolvedIsls(flowPath)
+        involvedLinks.size() == 1
+
+        when: "Update flow bandwidth to maximum link speed"
+        def linkSpeed = islUtils.getIslInfo(links, involvedLinks.first()).get().speed
+        flow.maximumBandwidth = linkSpeed
+        northboundService.updateFlow(flow.id, flow)
+
+        then: "Flow is really updated and has 'Up' status"
+        northboundService.getFlow(flow.id).maximumBandwidth == linkSpeed
+        Wrappers.wait(5) {
+            northboundService.getFlowStatus(flow.id).status == FlowState.UP
+        }
+
+        and: "The same link is used by updated flow"
```

Comment:
Looking at the code below, I think it's better to say 'the same path'

Reply:
Fixed.